harness/determined/exec/harness.py

```diff
@@ -95,6 +95,8 @@ def main(train_entrypoint: Optional[str]) -> int:
             distributed = _core.DistributedContext.from_horovod(horovod.hvd)
         elif distributed_backend.use_deepspeed():
             distributed = _core.DistributedContext.from_deepspeed()
+        elif distributed_backend.use_torch():
```

Comment:
Is it the intended behavior that if both horovod and torch are selected we use horovod?

Reply:
we don't support multiple dtrain launch layers. in theory this would never happen, harness is launched through a launch layer, and only the launch layer sets environment variables that determines this flag. i guess it's possible that somehow a user sets some of these environment variables themselves for some reason, but this is uncharted/unsupported behavior territory.